src/BorrowController.sol

```diff
@@ -125,22 +120,46 @@ contract BorrowController {
     }
 
     /**
-    @notice Reduces the daily limit used, when a user repays debt
-    @dev This is necessary to prevent a DOS attack, where a user borrows the daily limit and immediately repays it again.
-    @param amount Amount repaid in the market
-    */
+     * @notice Reduces the daily limit used, when a user repays debt
+     * @dev This is necessary to prevent a DOS attack, where a user borrows the daily limit and immediately repays it again.
+     * @param amount Amount repaid in the market
+     */
     function onRepay(uint amount) public {
-        uint day = block.timestamp / 1 days;
-        if(dailyBorrows[msg.sender][day] < amount) {
-            dailyBorrows[msg.sender][day] = 0;
-        } else {
-            //Safe to use unchecked, as dailyBorow is checked to be higher than amount
-            unchecked{
-                dailyBorrows[msg.sender][day] -= amount;
-            }
+        updateDailyBorrowLimit(msg.sender);
+        unchecked{
+            uint newLimit = remainingDailyBorrowLimit[msg.sender] + amount;
+            uint dailyLimit = dailyBorrowLimit[msg.sender];
+            remainingDailyBorrowLimit[msg.sender] = newLimit < dailyLimit ? newLimit : dailyLimit;
         }
     }
 
+    /**
+     * @notice Updates the daily borrow limit for the market
+     * @param market The address of the market which limit will be updated
+     */
+    function updateDailyBorrowLimit(address market) internal {
+        uint timeElapsed = block.timestamp - lastDailyBorrowLimitUpdate[market];
+        if(timeElapsed == 0) return;
+        // add daily limit capacity linearly based on time elapsed
+        uint dailyLimit = dailyBorrowLimit[market];
+        uint addedCapacity = timeElapsed * dailyLimit / 1 days;
+        uint newLimit = remainingDailyBorrowLimit[market] + addedCapacity;
+        // cap the limit to the maximum daily limit
+        remainingDailyBorrowLimit[market]= newLimit < dailyLimit ? newLimit : dailyLimit;
+        lastDailyBorrowLimitUpdate[msg.sender] = block.timestamp;
+    }
+
+    /**
+     * @notice Returns the available borrow limit of a market
+     * @param market The address of the market to return the available borrow limt
+     */
+    function availableBorrowLimit(address market) public view returns(uint) {
```

Comment:
Instead of duplicating code and risking inconsistency, I would suggest calling this function in `updateDailyBorrowLimit`.

Reply:
Removed `updateDailyBorrowLimit` in https://github.com/InverseFinance/FiRM/pull/96/commits/a6816168f19cdad2a5747f14745f8c5be3116eb5 and replaced with in-line code in `onRepay` and `onBorrow`